src/averageAges.js

```diff
@@ -56,6 +71,18 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  const women = people.filter(p => p.sex === 'f');
+
+  const determiner = (onlyWithSon)
+
+    ? people.filter(p => p.sex === 'm')
+      .filter(m => women.find(w => m.mother === w.name))
+      .map(s => s.born - women.find(w => s.mother === w.name).born)
+
+    : people.filter(p => women.find(w => p.mother === w.name))
+      .map(ch => ch.born - women.find(w => ch.mother === w.name).born);
+
+  return determiner.reduce((a, b) => a + b, 0) / determiner.length;
```

Comment:
a bit confusing to read to be honest
may be something like this
```suggestion
  const children = people.filter(person => (
    people.some(mother => mother.name === person.mother)
    && (onlyWithSon
      ? person.sex === 'm'
      : true
    )
  ))

  const ageDiffs = children
    .map(child => (
    	child.born - people.find(mother => mother.name === child.mother).born)
    )

  return ...;
```